src/validation/gltf/lightsPunctual/MaxarTemporalLightTraits/MaxarTemporalLightTraitsValidator.ts

```diff
@@ -0,0 +1,197 @@
+import { defined } from "3d-tiles-tools";
+
+import { ValidationContext } from "../../../ValidationContext";
+import { BasicValidator } from "../../../BasicValidator";
+
+import { GltfExtensionValidationIssues } from "../../../../issues/GltfExtensionValidationIssues";
+
+/**
+ * A class for validating the `MAXAR_temporal_light_traits` extension
+ * that extends `KHR_lights_punctual` lights.
+ *
+ * @internal
+ */
+export class MaxarTemporalLightTraitsValidator {
+  /**
+   * Validates the MAXAR_temporal_light_traits extension
+   *
+   * @param path - The path for validation issues
+   * @param extension - The temporal light traits extension object
+   * @param context - The validation context
+   * @returns Whether the extension is valid
+   */
+  static validateTemporalLightTraits(
+    path: string,
+    extension: any,
+    context: ValidationContext
+  ): boolean {
+    // Make sure that the given value is an object
+    if (
+      !BasicValidator.validateObject(
+        path,
+        "MAXAR_temporal_light_traits",
+        extension,
+        context
+      )
+    ) {
+      return false;
+    }
+
+    let result = true;
+
+    // Validate the flashing property (optional)
+    const flashing = extension.flashing;
+    if (defined(flashing)) {
+      if (
+        !MaxarTemporalLightTraitsValidator.validateFlashing(
+          path + "/flashing",
+          flashing,
+          context
+        )
+      ) {
+        result = false;
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Validates a flashing trait object
+   *
+   * @param path - The path for validation issues
+   * @param flashing - The flashing object
+   * @param context - The validation context
+   * @returns Whether the flashing trait is valid
+   */
+  private static validateFlashing(
+    path: string,
+    flashing: any,
+    context: ValidationContext
+  ): boolean {
+    // Make sure that the given value is an object
+    if (!BasicValidator.validateObject(path, "flashing", flashing, context)) {
+      return false;
+    }
+
+    let result = true;
+
+    // Validate the waveform property (required)
+    const waveform = flashing.waveform;
+    if (!defined(waveform)) {
+      const message = "The 'waveform' property is required";
+      const issue = GltfExtensionValidationIssues.INVALID_GLTF_STRUCTURE(
```

Comment:
I've seen that all/most of these are reporting a `INVALID_GLTF_STRUCTURE`. I think that most of them could/should use the same issue structures and mechanisms as the baseline 3D Tiles validator. For example, the `defined` check here and the manual `validWaveforms` check could be handled with the `validateEnum` function.